blocks/bricks/base.py

```diff
@@ -327,6 +328,16 @@ def copy_and_tag(variable, role, name):
         return unpack(outputs)
 
 
+# Ensure that Application instance are not pickled, and that when
+# unpickling, a reference to an existing Application object is
+# substituted instead of creating a new one.
```

Comment:
> Why doesn't it happen with regular pickling?

Because the Application object is created at interpret-time, but when pickled it's serialized like everything else, so when you unpickle there's two.